### PR TITLE
f-enhancement/gh-1117-enable-edit-of-active-year

### PR DESCRIPTION
**Pull request by fkis1**
Resolves: #1117

### CODE REVIEW

File: src/main/frontend/src/pages/administration/components/common/DrawerElements.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ const DrawerSelect = styled.select`
 `;
 
 export const WhiteBackgroundDiv = styled.div`
-  background-color: #fff;
+  background-color: ${props => (props.disabled ? '#f6f6f6' : '#fff')};
 `;
 
 const renderView = props => {
```
Comment (jolpazy) [inline src/main/frontend/src/pages/administration/components/common/DrawerElements.js]: destructure props

File: src/main/frontend/src/pages/administration/components/years/YearItem.js [hunk 2]
```diff
@@ -32,16 +32,13 @@ class YearItem extends Component {
...
-            !active
-              ? () =>
-                  this.props.openActionDrawer({
-                    type: 'editYear',
-                    heading: 'Edit Year',
-                    id
-                  })
-              : null
+          clickable
+          onClick={() =>
```
Comment (jolpazy) [inline src/main/frontend/src/pages/administration/components/years/YearItem.js]: extract onClick if possible

File: src/main/frontend/src/pages/administration/components/common/DrawerElements.js (hunk 1 above)
Comment (fkis1) [reply src/main/frontend/src/pages/administration/components/common/DrawerElements.js]: I have destructured props for the entire file

File: src/main/frontend/src/pages/administration/components/years/YearItem.js (hunk 2 above)
Comment (fkis1) [reply src/main/frontend/src/pages/administration/components/years/YearItem.js]: Extracted onClick function expressions to dedicated handlers.

File: src/main/frontend/src/pages/administration/components/years/EditYear.js [hunk 3]
```diff
@@ -149,7 +155,7 @@ class EditYear extends Component {
             <DrawerFooter
               isSubmitting={this.props.isSubmitting || values === initialValues}
               formClick={() => {
-                values.active
+                values.active && !isYearAlreadyActive
```
Comment (jolpazy) [inline src/main/frontend/src/pages/administration/components/years/EditYear.js]: check if ``formClick`` can be declared outside JSX.
you can pass things as ``setFieldTouched`` as parameter if it is extracted form the form

File: src/main/frontend/src/pages/administration/components/years/EditYear.js [hunk 4]
```diff
@@ -53,6 +53,28 @@ class EditYear extends Component {
...
 
+  handleSubmitInit = (
+    values,
+    errors,
+    isYearAlreadyActive,
+    setFieldTouched,
+    validateForm,
+    handleSubmit
+  ) => {
+    values.active && !isYearAlreadyActive
```
Comment (jolpazy) [inline src/main/frontend/src/pages/administration/components/years/EditYear.js]: when execution is being performed, in my opinion its nice r to use if statement

eg: ``if (thisIsTrue) {executeFunctionOne()} else {doSomethingElse()}``

use ternary for conditional outcome of a variable like: 
``timeLabel = timesStamp ? formatTime(timeStamp) : "time is not provided"``

**Pull request merged**